### ISSUE TITLE
Certain data no longer available via API

### ISSUE THREAD

**Issue by firewiremb** [openviess/PyViCare#359]
I have been using PyViCare for a long time. Until recently I used version 2.28.1. Since I noticed that the error PyViCareNotSupportedFeatureError('getHoursLoadClass1') was occurring in my log files, I carried out an update. I suspected that the API had changed and this might be fixed with a newer version.

After the update to version 2.32.0, not a single feature of the API was available anymore.

I was able to find out that version 2.31.0 is the last version that works for my circumstances. (The PyViCareNotSupportedFeatureError('getHoursLoadClass1') still occurs with this version.)

See also files in the appendix.

[dump.2.28.1.JSON](https://github.com/somm15/PyViCare/files/14157814/dump.2.28.1.JSON)
[dump.2.31.0.JSON](https://github.com/somm15/PyViCare/files/14157815/dump.2.31.0.JSON)
[dump.2.32.0.JSON](https://github.com/somm15/PyViCare/files/14157816/dump.2.32.0.JSON)

**Comment by CFenner** [comment]
I guess you have a Opto1 connected? Viessmann changed their api that the gateway is the first device to return in the list. Also with v32 the opto1 is handled as a gateway class and not as a hearing device. Make sure you access your heating device instead.

**Comment by firewiremb** [comment]
Thank you for your quick reply.
Yes, I have connected a Vitoconnect 100 OPTO1 to my heat pump.

I changed:

`device = vicare.devices[0]`

to

`device = vicare.devices[1]`

This solved my Problem using v32.

The `PyViCareNotSupportedFeatureError('getHoursLoadClass1')` however still occurs.
The last day this feature was available and providing data according to my log files was October 9, 2023.
Has this feature been disabled within the Viessmann API or is something wrong with my code?

**Comment by woehrl01** [comment]
@firewiremb yes, it's likely that viessmann disabled that data for your device. It could be that it moved to the paid tier (no guarantees).

**Comment by firewiremb** [comment]
Thank you for your answers.

**Comment by woehrl01** [comment]
Out of curiosity, it could be that the values have just moved. Do those numbers fit your previous ones? @firewiremb 

<img width="1082" alt="Bildschirmfoto 2024-02-05 um 09 02 03" src="https://github.com/somm15/PyViCare/assets/2535856/7a133ecd-7f60-44d2-957d-dc7a391159a6">

**Comment by firewiremb** [comment]
Thank you for inquiring again. No, unfortunately these are not the values I miss.
I found an older dump.json file that I created when the feature was still active. The values I'm missing are hoursLoadClassOne, ...Two, ...Three, ... and so on.

![grafik](https://github.com/somm15/PyViCare/assets/63193396/277d46d4-73fd-43d2-93e6-2112a84fb7c5)
[dump.json](https://github.com/somm15/PyViCare/files/14171650/dump.json)

**Comment by CFenner** [comment]
I suggest that you ask in the Viessmann community about the data point. The support may tell if they moved something.

**Comment by firewiremb** [comment]
I made a request in the Viessmann community:

https://community.viessmann.de/t5/Konnektivitaet/Daten-ueber-API-nicht-mehr-verfuegbar/m-p/425212#M53706

**Comment by lackas** [comment]
For future reference, PR #699 provides a cleaner solution to the device ordering issue - instead of hardcoding device indices, you can now filter by type:

```python
heating_device = next(d for d in vicare.devices if d.isHeating())
```

**Comment by lackas** [comment]
Regarding the missing `hoursLoadClassOne` data - PR #689 adds compressor load class methods with fallback logic that tries both `heating.compressors.0.statistics` and `heating.compressors.0.statistics.load` paths, as Viessmann moved this data on some devices.

**Comment by firewiremb** [comment]
Problem is solved with PR #689.
